godot-core/src/builtin/vectors/vector_macros.rs

```diff
@@ -251,94 +251,109 @@ macro_rules! impl_vector_index {
+            pub const LEFT: Self = Self::new(-1.0 as $Scalar, 0.0 as $Scalar);
 
-            /// Returns the vector scaled to unit length. Equivalent to `self / self.length()`. See
-            /// also `is_normalized()`.
-            ///
-            /// If the vector is zero, the result is also zero.
-            #[inline]
-            pub fn normalized(self) -> Self {
-                Self::from_glam(self.to_glam().normalize_or_zero())
-            }
+            /// Right unit vector. Represents the direction of right.
+            pub const RIGHT: Self = Self::new(1.0 as $Scalar, 0.0 as $Scalar);
+
+            /// Up unit vector. Y is down in 2D, so this vector points -Y.
+            pub const UP: Self = Self::new(0.0 as $Scalar, -1.0 as $Scalar);
+
+            /// Down unit vector. Y is down in 2D, so this vector points +Y.
+            pub const DOWN: Self = Self::new(0.0 as $Scalar, 1.0 as $Scalar);
         }
     };
 }
 
-/// Implements common constants and methods for integer type vectors. Works for any vector type that
-/// has `to_glam`, `from_glam` and `to_glam_real` functions.
-macro_rules! impl_integer_vector_glam_fns {
+/// Implements constants present on 3D vectors.
+macro_rules! impl_vector3x_consts {
     (
         // Name of the vector type.
         $Vector:ty,
         // Type of target component, for example `real`.
         $Scalar:ty
     ) => {
         impl $Vector {
-            /// Length (magnitude) of this vector.
-            #[inline]
-            pub fn length(self) -> $Scalar {
-                self.to_glam_real().length()
-            }
+            /// Left unit vector. Represents the local direction of left, and the global direction of west.
+            pub const LEFT: Self = Self::new(-1.0 as $Scalar, 0.0 as $Scalar, 0.0 as $Scalar);
 
-            /// A new vector with each component set to 1 if it's positive, -1 if it's negative,
-            /// and 0 if it's zero.
-            #[inline]
-            pub fn sign(self) -> Self {
-                Self::from_glam(self.to_glam().signum())
-            }
+            /// Right unit vector. Represents the local direction of right, and the global direction of east.
+            pub const RIGHT: Self = Self::new(1.0 as $Scalar, 0.0 as $Scalar, 0.0 as $Scalar);
+
+            /// Up unit vector.
+            pub const UP: Self = Self::new(0.0 as $Scalar, 1.0 as $Scalar, 0.0 as $Scalar);
+
+            /// Down unit vector.
+            pub const DOWN: Self = Self::new(0.0 as $Scalar, -1.0 as $Scalar, 0.0 as $Scalar);
+
+            /// Forward unit vector. Represents the local direction of forward, and the global direction of north.
+            pub const FORWARD: Self = Self::new(0.0 as $Scalar, 0.0 as $Scalar, -1.0 as $Scalar);
+
+            /// Back unit vector. Represents the local direction of back, and the global direction of south.
+            pub const BACK: Self = Self::new(0.0 as $Scalar, 0.0 as $Scalar, 1.0 as $Scalar);
```

Comment:
Shouldn't matter too much, but if we anyway have to cast between int/float, don't you want to go with integer literals directly? This would help readability a tiny bit:
```rs
pub const LEFT: Self = Self::new(-1 as $Scalar, 0 as $Scalar, 0 as $Scalar);
```

Reply:
Also, for 3D vectors, I actually prefer [our current docs](https://godot-rust.github.io/docs/gdext/master/godot/builtin/struct.Vector3.html#implementations). 

For example `FORWARD`, Godot writes:
> Represents the local direction of forward, and the global direction of north.

but this is imo clearer:
> Unit vector in -Z direction. Can be interpreted as “into the screen” in an untransformed 3D world.